tests/utils/test_modeling_utils.py

```diff
@@ -460,6 +462,59 @@ def test_model_from_config_torch_dtype_str(self):
         with self.assertRaises(ValueError):
             model = AutoModel.from_pretrained(TINY_T5, torch_dtype="int64")
 
+    def test_model_from_config_torch_dtype_composite(self):
+        """
```

Comment:
just missing testing with `torch.float16` as a torch dtype and not a string in the dict!